src/module_PI/Arduino_Uno/DataPacket.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+  dataPacket.h - A class designed around Serial Communication
+*/
+
+#include "DataPacket.h"
+#include "Arduino.h"
+#include "../lib/Cryptosuite/Sha/sha1.h"
+
+#define SIZE_OF_HASH_IN_PACKET 8
+#define SIZE_OF_DATAPACKET_IN_BYTES 64
+
+DataPacket::DataPacket(String messageString, String packetTypeString){
+  packetTypeString.toUpperCase();
+  this->messageString = messageString;
+  this->packetTypeString = packetTypeString;
+  appendPacketHeader();
+  appendMessage();
+}
+
+DataPacket::DataPacket(byte* packetContents){
+  for(int i = 0; i < SIZE_OF_DATAPACKET_IN_BYTES; i++){
+  packetHash[i] = packetContents[i];
+  }
+  parseSha1FromRawPacket(packetContents);
+  parsePacketTypeFromRawPacket(packetContents);
+  parseMessageFromRawPacket(packetContents);
+}
+
+byte* DataPacket::getPacketAsArray(){
+  return packetAsArray;
+}
+
+byte* DataPacket::getPacketHash(){
+  return packetHash;
+}
+
+int DataPacket::getSizeInBytes(){
+  return SIZE_OF_DATAPACKET_IN_BYTES;
+}
+
+String DataPacket::getPacketType(){
+  return packetTypeString;
+}
+
+String DataPacket::getMessage(){
+  return messageString;
+}
+
+void DataPacket::appendPacketHeader(){
+  appendHash();
+  appendPacketTypeByte();
+}
+
+void DataPacket::appendHash(){
+  Sha1.init();
+  Sha1.print(messageString);
+  for(int i = 0; i < SIZE_OF_HASH_IN_PACKET; i++){
+    packetHash[i] = Sha1.result()[i];
+  }
+  append(packetHash, SIZE_OF_HASH_IN_PACKET);
+}
+
+void DataPacket::appendPacketTypeByte(){
+  byte packetTypeByte = 0;
+  if(packetTypeString == "LOG"){
```

Comment:
Since these strings (LOG, COMMAND, etc) are used in multiple places throughout this same class these should probably be moved up as constants. That will protect from future refactors and remembering to change the spelling in one place but forgetting to change it in another. 

_(Or even better would be an enum of some sort but that might be more work than its worth)_